Objects/longobject.c

```diff
@@ -1174,7 +1146,7 @@ PyLong_FromLongLong(long long ival)
     v = _PyLong_New(ndigits);
     if (v != NULL) {
         digit *p = v->long_value.ob_digit;
-        Py_SET_SIZE(v, ival < 0 ? -ndigits : ndigits);
+        _PyLong_SetSignAndDigitCount(v, ival < 0 ? -1 : 1, ndigits);
```

Comment:
I don't think that a branchless version would buy us much here, if anything I think it would hurt legibility and be slightly worse in terms of performance.  Not only this is right after a call to `_PyLong_New()`, which dwarfs any bit fiddling we might want to do, this will most likely become a conditional move instruction, which is pretty efficient by itself.